### PR TITLE
Light client proxy support: fixes and integration tests

### PR DESCRIPTION
**Pull request by ashcherbakov**
- Added CLI integration tests for Light Client Proxy
- Support auto-obtain of trusted height and hash if they are not provided to `light` command
- Improved docs
- Model version fix: support `all-model-versions` command
- Workaround for an issue with EOF return from light client proxy from time to time (TODO: research in future)

### CODE REVIEW

File: cmd/dcld/cmd/light.go [hunk 1]
```diff
@@ -75,37 +95,43 @@ var (
...
-	LightCmd.Flags().StringVar(&trustLevelStr, "trust-level", "1/3",
+	LightCmd.Flags().StringVar(&trustLevelStr, FlagTrustLevel, "1/3",
 		"trust level. Must be between 1/3 and 3/3",
 	)
-	LightCmd.Flags().BoolVar(&sequential, "sequential", false,
+	LightCmd.Flags().BoolVar(&sequential, FlagSeq, false,
 		"sequential verification. Verify all headers sequentially as opposed to using skipping verification",
 	)
+	LightCmd.Flags().Int64Var(&startTimeout, FlagStartTimeout, 0,
+		"How many seconds to wait before staring the light client proxy. Mostly for test purposes when light client is started at the same time as the pool.")
```
Comment (spivachuk) [inline cmd/dcld/cmd/light.go:129]: Please correct the typo: staring -> starting

File: cmd/dcld/cmd/light.go [hunk 2]
```diff
@@ -160,6 +186,22 @@ func runProxy(cmd *cobra.Command, args []string) error {
...
 
+	// should set either both hash and height, or none of them
+	if trustedHeight == 0 && len(trustedHash) > 0 {
+		return fmt.Errorf("hash is set but height is not (--%s) ", FlagHeight)
+	}
+	if trustedHeight != 0 && len(trustedHash) == 0 {
+		return fmt.Errorf("height is set but hash is not (--%s) ", FlagHash)
+	}
+
+	// if trusted height and hash are not set, try toget it from primaries and witness
```
Comment (spivachuk) [inline cmd/dcld/cmd/light.go:197]: Please correct the typo: to get

File: integration_tests/light_client_proxy/auth.sh [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+sleep 5
+
+echo "Get $user account"
+result=$(execute_with_retry "dcld query auth account --address=$user_address")
+check_response "$result" "\"address\": \"$user_address\""
+
+test_divider
+
+
+# 5. check non-existent values when entry added via light client
```
Comment (spivachuk) [inline integration_tests/light_client_proxy/auth.sh:113]: What is meant under "when entry added via light client"? Entries cannot be added via light client, can they?

Comment (spivachuk) [reply integration_tests/light_client_proxy/auth.sh:113]: Should the comment be:
"check non-existent values via light client when entry added"?

File: integration_tests/light_client_proxy/auth.sh [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+result="$(bash -c "$cmd")"
+user_address=$(echo $passphrase | dcld keys show $user -a)
+user_pubkey=$(echo $passphrase | dcld keys show $user -p)
+
+random_string user2
+cmd="(echo $passphrase; echo $passphrase) | dcld keys add $user2"
+result="$(bash -c "$cmd")"
+user_address2=$(echo $passphrase | dcld keys show $user2 -a)
+
+# 1. check non-existent values when no entry added via light client
```
Comment (spivachuk) [inline integration_tests/light_client_proxy/auth.sh:15]: Should the comment be:
"check non-existent values via light client when no entry added"?

File: x/model/handler_test.go [hunk 5]
```diff
@@ -659,12 +767,12 @@ func NewMsgUpdateModel(signer sdk.AccAddress) *types.MsgUpdateModel {
 	}
 }
 
-func NewMsgCreateModelVersion(signer sdk.AccAddress) *types.MsgCreateModelVersion {
+func NewMsgCreateModelVersion(signer sdk.AccAddress, sv uint32) *types.MsgCreateModelVersion {
```
Comment (spivachuk) [inline x/model/handler_test.go:770]: It makes sense to use full field names in such boilerplate methods.

File: cmd/dcld/cmd/light.go (hunk 1 above)
Comment (ashcherbakov) [reply cmd/dcld/cmd/light.go:129]: Done

File: cmd/dcld/cmd/light.go (hunk 2 above)
Comment (ashcherbakov) [reply cmd/dcld/cmd/light.go:197]: Done

File: integration_tests/light_client_proxy/auth.sh (hunk 3 above)
Comment (ashcherbakov) [reply integration_tests/light_client_proxy/auth.sh:113]: Done

File: integration_tests/light_client_proxy/auth.sh (hunk 4 above)
Comment (ashcherbakov) [reply integration_tests/light_client_proxy/auth.sh:15]: Done

**Pull request closed without merging**